atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStore.java

```diff
@@ -0,0 +1,201 @@
+ * for multiple distinct sequences to coexist in one table.
+ *
+ * Within a single identified row, values are written with dynamic column keys as sequence numbers, which are
+ * positive longs. Furthermore, a {@link SequenceAndBound} which controls which of the values is considered to be
+ * the most recent in the coordination sequence and how long it is valid for is written at dynamic column key zero.
+ * This {@link SequenceAndBound} should contain the ID of the value that is considered to be the most current
+ * in the sequence.
+ */
+// TODO (jkong): Coordination stores should be able to clean up old values.
+public final class KeyValueServiceCoordinationStore implements CoordinationStore {
+    private static final Logger log = LoggerFactory.getLogger(KeyValueServiceCoordinationStore.class);
+
+    private static final TableMetadata COORDINATION_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription.Builder()
+                            .componentName("sequence")
+                            .type(ValueType.BLOB)
+                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                            .build())),
+            new ColumnMetadataDescription(new DynamicColumnDescription(
+                    NameMetadataDescription.create(
+                            ImmutableList.of(
+                                    new NameComponentDescription.Builder()
+                                            .componentName("sequenceNumber")
+                                            .logSafety(TableMetadataPersistence.LogSafety.SAFE)
+                                            .type(ValueType.VAR_LONG)
+                                            .build())),
+                    ColumnValueDescription.forType(ValueType.VAR_LONG))),
+            ConflictHandler.IGNORE_ALL,
+            TableMetadataPersistence.CachePriority.WARM,
+            false,
+            0,
+            false,
+            TableMetadataPersistence.SweepStrategy.NOTHING, // we do our own cleanup
+            false,
+            TableMetadataPersistence.LogSafety.SAFE);
+    private static final long WRITE_TIMESTAMP = 0L;
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMappers.newServerObjectMapper();
+
+    private final KeyValueService kvs;
+    private final byte[] coordinationSequence;
+
+    private KeyValueServiceCoordinationStore(KeyValueService kvs, byte[] coordinationSequence) {
+        this.kvs = kvs;
+        this.coordinationSequence = coordinationSequence;
+    }
+
+    public static CoordinationStore create(KeyValueService kvs, byte[] coordinationSequence) {
+        CoordinationStore coordinationStore = new KeyValueServiceCoordinationStore(kvs, coordinationSequence);
+        kvs.createTable(AtlasDbConstants.COORDINATION_TABLE, COORDINATION_TABLE_METADATA.persistToBytes());
```

Comment:
Not sure when we'll create this - will this fit in with out current async init code?

Reply:
As long as you call this from something that is being async initialized, yes